store-api-server/src/payment/service/payment.service.ts

```diff
@@ -403,6 +419,162 @@ WHERE nft_order.id = $1
+                let errors = error.response?.data?.errors;
+                if (errors && typeof errors == "object") {
+                    errorMessage = error.response?.data?.error + "---DETAILS:---" + JSON.stringify(errors);
+                }
+                console.log('get quote ERROR' + errorMessage);
+                throw new HttpException(
+                    `there is problem simplex api get quote please contact your backend services detail: (${errorMessage})`,
+                    HttpStatus.BAD_REQUEST
+                );
+            }
+        } else {
+            console.log("Unexpected error simplex api get quote");
+        }
+        throw new HttpException(
+            `there is problem simplex api get quote please contact your backend services`,
+            HttpStatus.BAD_REQUEST
+        );
+    }
+
+    const paymentId = uuidv4();
+    const orderId = uuidv4();
+
+    try {
+      var paymentResponse = await axios.post(
+          SIMPLEX_API_URL + "/wallet/merchant/v2/payments/partner/data",
+          {
+            account_details: {
+              app_provider_id: SIMPLEX_WALLET_ID,
+              app_version_id: "1.0.0",
+              app_end_user_id: "" + usr.id,
+              app_install_date: usr.createdAt ? new Date(usr.createdAt * 1000).toISOString() : new Date().toISOString(),
+              email: "example.cohen@simplex.com", // TODO NO WAY ?
+              phone: "+972509123456", // TODO NO WAY ?
+              signup_login: {
+                timestamp: new Date().toISOString(),
+                ip: "176.12.200.206" // TODO ?
+              }
+            },
+            transaction_details: {
+              payment_details: {
+                quote_id: quoteId,
+                payment_id: paymentId,
+                order_id: orderId,
+                destination_wallet: {
+                  currency: "USD-DEPOSIT",
+                  address: usr.userAddress,
+                  tag: ""
+                },
+                original_http_ref_url: "https://www.partner.com/" // TODO ?
+              }
+            }
+          },
+          {
+            headers: {
+              "Authorization": `ApiKey ${SIMPLEX_API_KEY}`
+            }
+          }
+      );
+
+    } catch (error) {
+        let errorMessage;
+        if (error instanceof Error) {
+            if (axios.isAxiosError(error) && error.response) {
+                errorMessage = error.response?.data?.error || error.response?.data;
+                let errors = error.response?.data?.errors;
+                if (errors && typeof errors == "object") {
+                    errorMessage = error.response?.data?.error + "---DETAILS:---" + JSON.stringify(errors);
+                }
+                console.log('payment req ERROR' + errorMessage);
+                throw new HttpException(
+                    `there is problem simplex api payment req please contact your backend services detail: (${errorMessage})`,
```

Comment:
Actually, these message does not reach the api callers. Above error handler catch the message and wrapped general info like: "current status is not available"

Reply:
Ok, then better to not throw an HttpException here. Just throw a string error I would say?

Because if at some point later on the part that catches this error is changed, we might start exposing these HttpExceptions you're throwing here with too detailed error messages.

Reply:
Ok , I removed error details from simplex api. But it will continue to throw http error when i get error from simplex api. Is it enough?